webapp/store/selectors/nav.js

```diff
@@ -0,0 +1,176 @@
+import assert from 'assert';
```

Comment:
this was previously the pluginMetadata selector, but made more sense as the nav selector now. Added a few more tools to this file for dealing with duplicates and nested paths. Cleaned up the selectors themselves as well.